src/init/features/genkit.ts

```diff
@@ -0,0 +1,66 @@
+import * as spawn from "cross-spawn";
+import { logger } from "../../logger";
+import { doSetup as functionsSetup } from "./functions";
+import { Options } from "../../options";
+import { Config } from "../../config";
+import { prompt } from "../../prompt";
+
+/**
+ * doSetup is the entry point for setting up the genkit suite.
+ */
+export async function doSetup(setup: any, config: Config, options: Options): Promise<void> {
+  if (setup.functions?.languageChoice !== "typescript") {
+    await prompt(setup, [
+      {
+        name: "continueFunctions",
+        type: "confirm",
+        message:
+          "Genkit with Firebase uses Cloud Functions for Firebase with TypeScript. Initialize Functions to continue?",
+        default: true,
+      },
+    ]);
+    if (!setup.continueFunctions) {
+      logger.info("Stopped Genkit initialization");
+      return;
+    }
+
+    // Functions with genkit should always be typescript
+    setup.languageOverride = "typescript";
+    await functionsSetup(setup, config, options);
```

Comment:
What about cases where I am not initing functions but I already am in a firebase directory with typescript functions? Does this work as you want?

Reply:
If you're already in a firebase directory with typescript functions, this will prompt you to initialize a new codebase because the default (`functions/`) is taken. This is expected and wanted; `genkit init` assumes you're in a clean node environment